products/health-care/digital-health-modernization/engineering/mobile-shared-dependency.md

```diff
@@ -0,0 +1,62 @@
+3. The proxy code in `/lib/sm`, `/lib/rx`, `lib/health_records` and any other MHV related directories remains as is and becomes a shared dependency of the two teams.<br/>
+  a. For accountability, we suggest adding a CODEOWNERS entry for each of these `lib` directories to ensure that **both** teams are required to approve any PRs that affect shared code.
+4. For now, keep any decisions about caching in the module-specific code (i.e. in either mobile or health apartment module), since each client may have different requirements here. Longer term, 
+5. For now, don't inline any capabilities around MHV account creation in the shared upstream integration library code. This code can be present in the shared library code, but it should be invoked explicitly by either the mobile app or VA.gov when needed. 
+
+```
+                          [team-specific]             [shared]
+                       ┌─────────────────────────────────────────┐
+                       │                                         │
+                       │ ┌─────────────────┐                     │
+                       │ │                 │                     │  ┌───────────┐
+┌────────────────┐     │ │ Mobile module   │                     │  │           │
+│                │     │ │ (controllers+   │          ┌───────┐  │  │           │
+│   Mobile App   ├─────┤►│  models)        │  ┌──────►│lib/sm ├──┼─►│           │
+│                │     │ │                 │  │       └───────┘  │  │           │
+└────────────────┘     │ │                 ├──┤                  │  │           │
+                       │ │                 │  │       ┌───────┐  │  │           │
+                       │ └─────────────────┘  ├──────►│lib/rx ├──┼─►│           │
+                       │                      │       └───────┘  │  │           │
+                       │ ┌─────────────────┐  │                  │  │ MHV APIs  │
+                       │ │                 │  │    ┌──────────┐  │  │           │
+                       │ │ MyHealth module │  ├───►│lib/mhv_ac├──┼─►│ (upstream)│
+┌────────────────┐     │ │ (controllers+   ├──┤    └──────────┘  │  │           │
+│                │     │ │  models)        │  │                  │  │           │
+│  VA.gov        ├─────┤►│                 │  │  ┌────────────┐  │  │           │
+│                │     │ │ [currently in   │  └─►│lib/records ├──┼─►│           │
+└────────────────┘     │ │  main rails app]│     └────────────┘  │  │           │
+                       │ │                 │                     │  │           │
+                       │ └─────────────────┘                     │  └───────────┘
+                       │                                         │
+                       │           vets-api                      │
+                       └─────────────────────────────────────────┘
+
+```
+## Questions
+
+1. **Would it be better to use the same API endpoints for both clients?**
```

Comment:
In an ideal world I'd like one API but agree that auth and different models/schemas are a big enough issue to not do it. Something like GraphQL or a [REST alternative](https://www.graphiti.dev/guides/) could solve that. But, as you mention, there's the (pre) caching issue too which only gets more complicated when custom schemas can be delivered.